homeassistant/components/shelly/entity.py

```diff
@@ -320,3 +361,84 @@ def device_state_attributes(self) -> dict:
             return None
 
         return self.description.device_state_attributes(self.wrapper.device.status)
+
+
+class ShellySleepingBlockAttributeEntity(ShellyBlockAttributeEntity, RestoreEntity):
+    """Represent a shelly sleeping block attribute entity."""
+
+    # pylint: disable=super-init-not-called
+    def __init__(
+        self,
+        wrapper: ShellyDeviceWrapper,
+        block: aioshelly.Block,
+        attribute: str,
+        description: BlockAttributeDescription,
+        entry: Optional[ConfigEntry] = None,
+    ) -> None:
+        """Initialize the sleeping sensor."""
+        self.wrapper = wrapper
+        self.entry = entry
+        self.restored_state = None
+
+        if wrapper.restored_device:
```

Comment:
Isn't this always a restored device because this is the SleepingBlockAttributeEntity?

Reply:
Revised the logic here, removed `restored_device`, logic based on `block is None`